### DISCUSSION TITLE
Feature Request: Realistic Airplane Physics

### DISCUSSION THREAD

**Discussion by AnomolyAvery** [SmartlyDressedGames/Unturned-3.x-Community#4657] [Ideas]
**Description:**
I'd like to request a more realistic approach to airplane physics in Unturned. The current flight mechanics, while sufficient for basic gameplay, could be significantly improved to provide a more authentic and engaging experience, particularly for players interested in aviation.

**Current Issues:**
1. **Glide and Aerodynamic Mechanics:** When the engine is cut off mid-air or air speed is reduced, the airplane does not glide as it should. Instead, it falls slowly and unnaturally, which lacks realism. Additionally, the physics system does not seem to account for aerodynamic forces such as drag, lift, and gravity realistically. This results in planes behaving more like helicopters when descending, rather than following a natural flight path.

2. **Stalling:** Airplanes in Unturned do not exhibit realistic stalling behavior. In real-life aviation, stalling occurs when the aircraft's wings lose lift due to slow speed or an excessively high angle of attack. In the game, this mechanic is either missing or underrepresented.

3. **Landing Dynamics:** Landing an airplane feels overly simplified. The impact of speed, angle, and descent rate on the landing outcome is minimal, which reduces the challenge and realism of piloting aircraft.

4. **Wind, Turbulence, and Weather Effects:** The absence of wind and turbulence effects makes flying predictable and less immersive. Although the game features a weather system with rain and thunderstorms, these conditions do not seem to affect airplane flight dynamics. Incorporating weather effects into the flight experience would add significant depth and realism.

**Suggestions for Improvement:**
1. **Implement Realistic Glide Dynamics:** Introduce mechanics that allow airplanes to glide based on their current speed, altitude, and angle when the engine is turned off or speeds are reduced. Improve the simulation of aerodynamic forces to ensure that airplanes behave naturally, responding to changes in speed, angle, and atmospheric conditions.

2. **Accurate Stalling Mechanics:** Implement a stalling system where airplanes lose lift and start to descend rapidly if they slow down too much or are flown at an incorrect angle. This would make flight more challenging and rewarding.

3. **Advanced Landing Mechanics:** Make landings more dependent on speed, angle, and descent rate. This would encourage players to learn proper landing techniques and add to the realism of piloting aircraft in Unturned.

4. **Enhanced Weather Interaction:** Integrate wind, turbulence, and weather effects into the flight physics. For example, rain and thunderstorms could increase the likelihood of turbulence, and impact the airplane's control and performance. These elements would make flying less predictable and require players to adapt to changing conditions.

**Benefits:**
- Provides a more immersive and realistic flight experience.
- Enhances the challenge and skill required for piloting aircraft.
- Appeals to aviation enthusiasts and those seeking more depth in aerial gameplay.
- Improves overall gameplay dynamics, making flying more rewarding and less predictable.

**Conclusion:**
By addressing these issues, the airplane physics in Unturned could be elevated to offer a more authentic and challenging experience. These improvements would greatly enhance the game for players who enjoy flying and would add significant value to the overall gameplay. Thank you for considering this feature request!

**Comment by SDGNelson** [comment]
Thanks for moving into a discussion post! I agree it would fantastic to flesh out the airplane physics. Ideally, I intend to work on them as part of the vehicle focus this year, but I'm not 100% sure what the order of work will be. My first priority was wheeled vehicles since they were the worst, and the next big step on that path after updating two-wheelers this week is vehicles with crawler tracks like tanks.

**Reply by AnomolyAvery** [reply]
Thank you for considering my feature request! I'm glad to hear that you're planning to work on improving airplane physics as part of the vehicle focus this year. I understand the need to prioritize wheeled vehicles first, and I appreciate the work you're putting into enhancing the overall vehicle experience in Unturned.

When you get to the point of addressing airplane physics, I’d be more than happy to help in any way I can, including providing calculations, simulations, or development insights related to glide dynamics, stalling behavior, and aerodynamic forces; along with assisting in testing new mechanics. I'm truly excited about the potential for a more realistic flight experience in the game!

Looking forward to seeing how it develops!

**Reply by SDGNelson** [reply]
Thank you very much for your offer and kind words, I appreciate it - I'm sure I have a lot to learn about airplane physics, so I may need to take you up on it when the time comes! :)
